packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -111,37 +120,78 @@ const ImageLoader = {
       if (typeof failure === 'function') {
         failure();
       }
-      ImageLoader.abort(requestId);
+      ImageLoader.release(requestId);
       clearInterval(interval);
     }
   },
-  has(uri: string): boolean {
+  has(uri: ?string): boolean {
     return ImageUriCache.has(uri);
   },
-  load(uri: string, onLoad: Function, onError: Function): number {
+  load(
+    source: ImageSource,
+    onLoad: (ImageResult) => void,
+    onError: Function
+  ): number {
     id += 1;
     const image = new window.Image();
-    image.onerror = onError;
-    image.onload = (e) => {
+
+    const handleLoad = () => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
-      if (typeof image.decode === 'function') {
-        // Safari currently throws exceptions when decoding svgs.
-        // We want to catch that error and allow the load handler
-        // to be forwarded to the onLoad handler in this case
-        image.decode().then(onDecode, onDecode);
-      } else {
-        setTimeout(onDecode, 0);
-      }
+      const onDecode = () => {
+        ImageUriCache.add(source.uri);
+        onLoad({
+          source: {
+            uri: image.src,
+            width: image.naturalWidth,
+            height: image.naturalHeight
+          }
+        });
+      };
+
+      // Safari currently throws exceptions when decoding svgs.
+      // We want to catch that error and allow the load handler
+      // to be forwarded to the onLoad handler in this case
+      image.decode().then(onDecode, onDecode);
     };
-    image.src = uri;
-    requests[`${id}`] = image;
+
+    image.onerror = onError;
+    image.onload = handleLoad;
+    requests[id] = { image, source };
+
+    // It's not possible to use headers with `image.src`, that's why we use a different strategy for sources with headers
+    if (source.headers) {
+      const abortCtrl = new AbortController();
```

Comment:
Maybe a code comment would help here, but is the reason for the abort controller so that if the component unmounts before the image is loaded, then it cleans up a memory leak?

Reply:
Yes and more

If the Image component starts a new request, the old one is discarded - we want to stop fetching it, and focus on the new request 
Same thing if the component unmounts, we want to stop loading 

There's an `ImageLoader.abort` method, I think any description about why it exists and when to use it should be there 
The specific code here is just so the `ImageLoader.abort` method can stop the newly added code working with `fetch`